maas/client/flesh/tests/test_machines.py

```diff
@@ -101,3 +102,36 @@ def test_calls_handler_with_hostnames(self):
         options = subparser.parse_args(hostnames)
         cmd.execute(origin, options, target=tabular.RenderTarget.yaml)
         origin.Machines._handler.read.assert_called_once_with(hostname=hostnames)
+
+
+class TestMachine(TestCaseWithProfile):
+    """Tests for `cmd_machine`."""
+
+    def test_machine_details_with_tags(self):
+        origin = make_origin()
+        parser = ArgumentParser()
+        machine_objs = [
+            {
+                "hostname": make_name_without_spaces(),
+                "architecture": "amd64/generic",
+                "status": NodeStatus.READY.value,
+                "status_name": NodeStatus.READY.name,
+                "owner": None,
+                "power_state": PowerState.OFF.value,
+                "cpu_count": 2,
+                "memory": 1024,
+                "pool": {"id": 1, "name": "pool1", "description": "pool1"},
+                "zone": {"id": 1, "name": "zone1", "description": "zone1"},
+                "tag_names": ["tag1", "tag2"],
+                "distro_series": "",
+                "power_type": "Manual",
+            },
+        ]
+        origin.Machines._handler.read.return_value = machine_objs
+        cmd = machines.cmd_machine(parser)
+        subparser = machines.cmd_machine.register(parser)
+        options = subparser.parse_args([machine_objs[0]["hostname"]])
+        output = yaml.safe_load(
+            cmd.execute(origin, options, target=tabular.RenderTarget.yaml)
+        )
+        self.assertEqual(output.get("tags"), "tag1, tag2")
```

Comment:
Wouldn't it make sense for the yaml renderer to render the tags as a list of strings?

Reply:
+1

Reply:
Good point, fixed!